terraform/task_definition.tf

```diff
@@ -0,0 +1,41 @@
+resource "aws_ecs_task_definition" "singer_task" {
+  family                   = "singer-strokepredictor"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = "1024"         # 1 vCPU
+  memory                   = "2048"         # 2 GB
+  execution_role_arn       = aws_iam_role.ecs_execution_role.arn
+  task_role_arn            = aws_iam_role.ecs_task_role.arn
+
+  ephemeral_storage {
+    size_in_gib = 50
+  }
+
+  container_definitions = jsonencode([
+    {
+      name      = "singer-container"
+      image     = "061051226319.dkr.ecr.us-east-1.amazonaws.com/singer_strokepredictor:latest"
```

Comment:
you should be able to reference your ECR Terraform resource directly instead of hard-coding this value

```
image = "${aws_ecr_repository.singer_strokepredictor.repository_url}:latest"
```